distributions/aws/test/e2e/tests/test_storage_efs.py

```diff
@@ -66,3 +71,43 @@ def test_pvc_with_volume(
         claim_name = static_provisioning["claim_name"]
         claim_list = subprocess.check_output("kubectl get pvc -A".split()).decode()
         assert claim_name in claim_list
+
+
+class TestEFS_Dynamic:
+    @pytest.fixture(scope="class")
+    def setup_dynamic(self, metadata, kustomize, dynamic_provisioning):
+        metadata_file = metadata.to_file()
+        print(metadata.params)  # These needed to be logged
+        print("Created metadata file for TestSanity", metadata_file)
+
+    def test_pvc_with_volume_dynamic(
+        self,
+        metadata,
+        setup_dynamic,
+        account_id,
+        create_efs_volume,
+        dynamic_provisioning,
+    ):
+        driver_list = subprocess.check_output("kubectl get csidriver".split()).decode()
+        assert "efs.csi.aws.com" in driver_list
+
+        pod_list = subprocess.check_output("kubectl get pods -A".split()).decode()
+        assert "efs-csi-controller" in pod_list
+
+        sa_account = subprocess.check_output(
+            "kubectl describe -n kube-system serviceaccount efs-csi-controller-sa".split()
+        ).decode()
+        assert f"arn:aws:iam::{account_id}:role" in sa_account
+
+        fs_id = create_efs_volume["file_system_id"]
+        assert "fs-" in fs_id
+
+        efs_claim_dyn = dynamic_provisioning["efs_claim_dyn"]
+        claim_list = subprocess.check_output("kubectl get pvc -A".split()).decode()
+        assert efs_claim_dyn in claim_list
+
+        # TODO: Add this check to static provisioning as well
+        # TODO: Status is currently pending but we should also be testing for this to reach bound state
```

Comment:
Yes without bound state the test is incomplete.

Reply:
Yes that is a planned enhancement for this test already in the works.

Reply:
This change is now a part of the PR.